kernel/src/main/java/org/kframework/kprove/KProveOptions.java

```diff
@@ -55,4 +54,12 @@ public synchronized File specFile(FileUtil files) {
 
     @Parameter(names="--depth", description="The maximum number of computational steps to prove")
     public Integer depth;
+
+    @Parameter(names = "--boundary-cells", description = "The comma-separated list of cells used to mark the boundary " +
+            "of evaluation. If option is specified, execution ends when these cells in the current term match same " +
+            "cells in the target term. (Except for step 1, for which boundary checking is disabled.)" +
+            "If the whole current term matches target, execution is successful. " +
+            "Otherwise it fails on the present path. If option is not specified, full target term implication is checked " +
+            "on every step. Recommended value for KEVM: \"k,pc\"")
```

Comment:
Explanation of what happens in Java backend:
When the pattern matches, it generates a substitution with 2 kinds of variables: 

1. Cell body for cells I mentioned in `--boundary-cells`. These vars will have the name starting with `BOUND_`. These have to be preserved.
2. The rest of the config - cells I don't care about. These will be anonymous vars with name starting with `_`. In function `getBoundarySubstitution` I only keep the vars in 1st category.

For each step, after I get the clean substitution for current term and target term, I consider them to match if the equality `currentMatch == targetMatch` is not false. That is, if the match contains symbolic terms, and equality is neither true nor false, `boundaryCellsMatchTarget` will be true. That is because the option is not designed to be used in conjuction with symbolic reasoning. Matching should not depend on symbolic values.

I suppose this fragment is most complex to understand, the point where result of boundary matching is actually used:
https://github.com/kframework/k/blob/0b7cbaa63d4ca14c8fd6530ec8c62f82c050dac1/java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java#L648-L668

All 3 cases are covered by the tests, and explained in the description of the option.

Reply:
This seems complicated enough to warrant comments inline in the boundaryCellsMatchTarget and getBoundarySubstitution methods to me.

Reply:
> 
> 
> you should not be talking about kevm in the command line flags of k.

Done.

Reply:
> 
> 
> This seems complicated enough to warrant comments inline in the boundaryCellsMatchTarget and getBoundarySubstitution methods to me.

Done.